lib/core/include/irods/irods_server_properties.hpp

```diff
@@ -148,13 +148,32 @@ namespace irods
 
         void remove( const std::string& _key );
 
-        /// \brief Get a reference to the underlying json object. Writing
-        /// in particular should be done with caution as there is no guarantee
-        /// that this will be safe.
-        /// In threaded/asynchronous contexts, this function bypasses safety guarantees.
-        nlohmann::json& map() noexcept
+    private:
+        class [[nodiscard("Do not discard lock!")]] config_handle
+        {
+          public:
+            config_handle(const nlohmann::json& json, std::shared_mutex& mutex)
+                : server_config{json}, lock{mutex}
+            {
+            }
+
+            config_handle(const config_handle&) = delete;
+            auto operator=(const config_handle&) -> config_handle& = delete;
+
+            [[nodiscard]] auto get_json() const & noexcept -> const nlohmann::json& {
+                return server_config;
+            }
+
+          private:
+            const nlohmann::json& server_config;
+            std::shared_lock<std::shared_mutex> lock;
+        };
+
+    public:
+        /// \brief Acquire a config_handle object containing a const reference to the underlying json object.
```

Comment:
Add words explaining what `config_handle` is for. You can include example code if you want. There's a few functions in the codebase that provide example code. See irods_logger.hpp for an example.

Missing `\since` property.